build-tools/scripts/jenkinsfile

```diff
@@ -0,0 +1,374 @@
+                        )
+                        plot(
+                            title: 'Runtime merged',
+                            csvFileName: 'plot-runtime-merged-test-times.csv',
+                            csvSeries: [[
+                                displayTableFlag: true, file: 'TestResult-Mono.Android_Tests-times.csv', inclusionFlag: 'OFF'
+                            ]],
+                            group: 'Tests times', logarithmic: true, style: 'line', yaxis: 'ms'
+                        )
+                        plot(
+                            title: 'Xamarin.Forms app startup',
+                            csvFileName: 'plot-xamarin-forms-startup-test-times.csv',
+                            csvSeries: [[
+                                displayTableFlag: true, file: 'TestResult-Xamarin.Forms_Test-times.csv', inclusionFlag: 'OFF'
+                            ]],
+                            group: 'Tests times', logarithmic: true, style: 'line', yaxis: 'ms'
+                        )
+                        plot(
+                            title: 'Xamarin.Forms app',
+                            csvFileName: 'plot-xamarin-forms-tests-size.csv',
+                            csvSeries: [[
+                                displayTableFlag: true, file: 'TestResult-Xamarin.Forms_Tests-values.csv', inclusionFlag: 'OFF'
+                            ]],
+                            group: 'Tests size', logarithmic: true, style: 'line', yaxis: 'ms'
+                        )
+
+                        plot(
+                            title: 'Hello World',
+                            csvFileName: 'plot-hello-world-build-times.csv',
+                            csvSeries: [[
+                                displayTableFlag: true, file: 'TestResult-Timing-HelloWorld.csv', inclusionFlag: 'OFF'
+                            ]],
+                            group: 'Build times', logarithmic: true, style: 'line', yaxis: 'ms'
+                        )
+
+                        plot(
+                            title: 'Xamarin.Forms',
+                            csvFileName: 'plot-xamarin-forms-integration-build-times.csv',
+                            csvSeries: [[
+                                displayTableFlag: true, file: 'TestResult-Timing-Xamarin.Forms-Integration.csv', inclusionFlag: 'OFF'
+                            ]],
+                            group: 'Build times', logarithmic: true, style: 'line', yaxis: 'ms'
+                        )
+                    }
+                }
+            }
+        } catch (error) {
+            echo "ERROR : Plot build & test metrics: Unexpected error: ${error}. Marking build as UNSTABLE"
+            currentBuild.result = 'UNSTABLE'
+        } finally {
+            echo "Stage result: Plot build & test metrics: ${currentBuild.currentResult}"
+        }
+
+        stageStatus = 0
+        try {
+            stage('Upload test results') {
```

Comment:
This stage name seems odd; it says "Upload", but I don't see a `publishPackages()` call.  Perhaps this should be **Render test results**?

Reply:
Per commit https://github.com/xamarin/xamarin-android/pull/2603/commits/894685c210088a8f935e5e0c15bc7ef46ad01c18 we decided to with 'Publish test results' consistent with other stages in this file relating to the publishing of files or results